sdk/highlight-py/pyproject.toml

```diff
@@ -31,13 +31,14 @@ django = { version = ">=3", optional = true }
 flask = { version = ">=2", optional = true }
 fastapi = { version = ">=0", optional = true }
 uvicorn = {version = ">=0", extras = ["standard"], optional = true }
-opentelemetry-api = "1.20.0"
-opentelemetry-distro = { extras = ["otlp"], version = "0.41b0" }
-opentelemetry-exporter-otlp-proto-http = "1.20.0"
-opentelemetry-instrumentation = "0.41b0"
-opentelemetry-instrumentation-logging = "0.41b0"
-opentelemetry-proto = "1.20.0"
-opentelemetry-sdk = "1.20.0"
+opentelemetry-api = "1.21.0"
+opentelemetry-distro = { extras = ["otlp"], version = "0.42b0" }
+opentelemetry-exporter-otlp-proto-http = "1.21.0"
+opentelemetry-instrumentation = "0.42b0"
+opentelemetry-instrumentation-logging = "0.42b0"
+opentelemetry-proto = "1.21.0"
+opentelemetry-sdk = "1.21.0"
+opentelemetry-instrumentation-requests = "0.42b0"
```

Comment:
in a future pr, we can break these auto instrumentations into their own extras that determine whether the instrumentation is auto enabled. no need to do that here tho